### DISCUSSION TITLE
Do we need to load the ingredient taxonomy for Open Food Facts ?

### DISCUSSION THREAD

**Discussion by teolemon** [openfoodfacts/openfoodfacts-androidapp#3980] [Q&A]
Do we need to load the ingredient taxonomy for Open Food Facts ?

https://github.com/openfoodfacts/openfoodfacts-androidapp/blob/develop/app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/LoadTaxonomiesWorker.kt#L50

**Comment by VaiTon** [comment]
If I remember correctly, we use the ingredient taxonomy to analyze ingredients. Am I wrong?

**Comment by VaiTon** [comment]
Also, this is done in a Background Worker so it should not cause any ANR

**Comment by teolemon** [comment]
I believe it's not used in Open Food Facts, only in Open Beauty Facts (for the ingredient tab).
We also merged some foundation work by dobriseb, but it's not actively called in either of the apps.
Also, I do have an ANR when first opening the app (during the onboarding slides), which is the time the download happens.

**Comment by VaiTon** [comment]
> Also, I do have an ANR when first opening the app (during the onboarding slides), which is the time the download happens.

But this is not related to downloads. They _should_ be downloaded in background and we _should not_ wait for them. Does this happen on develop or prod?

**Comment by naivekook** [comment]
just curious what is `taxonomy`?
